cmd/zpool/zpool_main.c

```diff
@@ -9728,12 +9728,15 @@ wait_status_thread(void *arg)
 		} else {
 			timeout.tv_nsec = nanos;
 		}
-
-		if (sem_timedwait(&wd->wd_sem, &timeout) == 0) {
+		pthread_mutex_lock(&wd->wd_mutex);
+		int ret = pthread_cond_timedwait(&wd->wd_cv, &wd->wd_mutex,
+		    &timeout);
+		pthread_mutex_unlock(&wd->wd_mutex);
+		if (ret == 0) {
 			break; /* signaled by main thread */
-		} else if (errno != ETIMEDOUT) {
-			(void) fprintf(stderr, gettext("sem_timedwait failed: "
-			    "%s\n"), strerror(errno));
+		} else if (errno != ETIMEDOUT && errno != 316) {
```

Comment:
I believe that `pthread_cond_timedwait` uses its return value rather than `errno` to specify the error that occurred.

Reply:
Ah so it does! That clears up that mystery.